woocommerce/woocommerce
# Fix the inconsistent formatting of the numeric columns in the Leaderboards on the Analytics > Dashboard page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #36748

This PR fixes the issue by:

- Add the `format` property to the items of the `rows` array in Analytics' Leaderboards API response to indicate how it's formatted.
- Make the `Leaderboard` component within Analytics aware of the currency context and format the numeric columns accordingly.

Extra changes to the *client/analytics/components/leaderboard/test/index.js* file:

- Apply better practice
- Avoid a warning when running JS tests
   ![3](https://github.com/woocommerce/woocommerce/assets/17420811/87ea3cc2-f46d-4efe-9986-d9087d52fc4f)
   Ref: https://github.com/woocommerce/woocommerce/blob/9.0.2/packages/js/components/src/pagination/pagination.tsx#L44

### Screenshots:

#### 📷 Using store currency options

![1](https://github.com/woocommerce/woocommerce/assets/17420811/cab23874-80f2-4da4-aeff-cdf29e82e68e)

#### 📷 Using filtered currency options

![2](https://github.com/woocommerce/woocommerce/assets/17420811/5c07cbff-86bd-49d4-8661-9437d539a84c)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this PR, and run `pnpm install` and `pnpm build`
2. Open another webpage to view the current currency options on the **WooCommerce > Settings > General** page
3. Go to the **Analytics > Overview** page
4. If there are not large enough values to test, modifying these two lines locally can adjust the API response
   - https://github.com/woocommerce/woocommerce/blob/45022b3705b3b5751870e09cd8603ea1e58942ea/plugins/woocommerce/src/Admin/API/Leaderboards.php#L203
   - https://github.com/woocommerce/woocommerce/blob/45022b3705b3b5751870e09cd8603ea1e58942ea/plugins/woocommerce/src/Admin/API/Leaderboards.php#L208
5. Check if the currency options are applied to the currency and formatting of all amount values in the Leaderboards
6. Check if the thousand separator of the currency options is applied to all values of the Items Sold column in the Leaderboards
7. Add the following PHP code snippet, where the currency options should be different from the ones in step 2:
   ```php
   add_action( 'admin_enqueue_scripts', function () {
   	wp_add_inline_script( 'wp-hooks', <<<EOD
   	wp.hooks.addFilter( 'woocommerce_admin_report_currency', 'dev-blog-example', function() {
   		return {
   			code: "NZD",
   			decimalSeparator: ".",
   			precision: 2,
   			priceFormat: "%1\$s %2\$s",
   			symbol: "\$NZ",
   			symbolPosition: "left",
   			thousandSeparator: ",",
   		};
   	} );
   	EOD );
   } );
   ```
8. Repeat the test similar to steps 5 and 6, but check that the formatting of the values reflects the currency options returned by the filter in step 7.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
